### ISSUE TITLE
Missing validation error on DS compilation with no patch constant signature

### ISSUE THREAD

**Issue by jenatali** [microsoft/DirectXShaderCompiler#6629]
**Description**
D3D12 shader linkages require identical declarations of the patch constant signature between HS and DS. Since HS must output tessellation levels, DS must therefore input them. FXC had such an error:
> error X3502: ds_5_0 tessfactor inputs missing

**Steps to Reproduce**
```hlsl
[domain("tri")]
float4 DSMain(float4 pos : POS) : SV_Position
{
	return pos;
}
```
Compile with FXC target `ds_5_0` and you get error X3502. DXC succeeds. Attempting to use such a shader results in a D3D12 PSO creation failure (`E_FAIL`) with no debug layer error message (D3D bug 50951962 is tracking this missing error message).

See https://godbolt.org/z/5bs15vx4s for DXC on trunk, and https://shader-playground.timjones.io/3b46dc2f059f0b0611dcf44c264ccaab for FXC.

**Environment**
- DXC version: `dxc(private) 1.8.0.10001 (main, ff623f8a)`

**Comment by jenatali** [comment]
Update: The shader I posted above had a patch constant signature (and no per-control-point inputs). This one has none: https://godbolt.org/z/49j39v46v.

**Comment by damyanp** [comment]
@jenatali - setting this to dormant for now.  Is there any urgency to this or was it filed more as an FYI?

**Comment by jenatali** [comment]
Nah, just an FYI. It found a D3D bug - using the resulting shader caused failure with no debug messages indicating why it failed. We're adding a debug message, but figured you might want to fix the compiler too.
